torch/_dynamo/variables/builtin.py

```diff
@@ -1212,9 +1212,18 @@ def _grad_changed(old, new):
                             else:
                                 grapharg.example.grad = None
                         return VariableBuilder(tx, source)(grapharg.example.grad)
-                unimplemented("tensor grad")
+
+                # No match for real value in inputs, fall back to
+                # var_getattr, which is sound (May produce a GetAttrVariable)
+                try:
+                    return obj.var_getattr(tx, name).clone(source=source)
```

Comment:
Putting a `getattr` that returns a tensor in the graph feels a bit sus to me (functionalization will miss any mutations that happen to that tensor) . Is there a reason we can't:

(1) guard on whether or not the .grad field is None at the time that we started tracing

(2) If the `.grad` field was a valid tensor at trace time, lift it into a new input to the graph, otherwise burn a `None` into the graph.

Reply:
Yeah if we don't want getattr in the graph, this won't work.

(2) Lifting in a non source case isn't really possible? Maybe what we want is a get_grad native op?